src/Microsoft.Diagnostics.DebugServices.Implementation/ModuleServiceFromDataReader.cs

```diff
@@ -68,41 +68,40 @@ public override ImmutableArray<byte> BuildId
                 }
             }
 
-            public override VersionData VersionData
+            public override VersionData GetVersionData()
             {
-                get 
+                if (InitializeValue(Module.Flags.InitializeVersion))
                 {
-                    if (InitializeValue(Module.Flags.InitializeVersion))
+                    if (_moduleInfo.Version != EmptyVersionInfo)
                     {
-                        if (_moduleInfo.Version != EmptyVersionInfo)
-                        {
-                            _versionData = _moduleInfo.Version.ToVersionData();
-                        }
-                        else
+                        _versionData = _moduleInfo.Version.ToVersionData();
+                    }
+                    else
+                    {
+                        if (_moduleService.Target.OperatingSystem != OSPlatform.Windows)
                         {
-                            if (_moduleService.Target.OperatingSystem != OSPlatform.Windows)
-                            {
-                                _versionData = GetVersion();
-                            }
+                            _versionData = GetVersion();
                         }
                     }
-                    return _versionData;
                 }
+                return _versionData;
             }
 
-            public override string VersionString
+            public override string GetVersionString()
             {
-                get
+                if (InitializeValue(Module.Flags.InitializeProductVersion))
                 {
-                    if (InitializeValue(Module.Flags.InitializeProductVersion))
+                    if (_moduleService.Target.OperatingSystem != OSPlatform.Windows && !IsPEImage)
                     {
-                        if (_moduleService.Target.OperatingSystem != OSPlatform.Windows && !IsPEImage)
-                        {
-                            _versionString = _moduleService.GetVersionString(ImageBase);
-                        }
+                        _versionString = _moduleService.GetVersionString(this);
                     }
-                    return _versionString;
                 }
+                return _versionString;
+            }
+
+            public override string LoadSymbols()
+            {
+                return _moduleService.SymbolService.DownloadSymbolFile(this);
```

Comment:
Yes - not something for this PR, but this will make symbol loading really slow for production apps (not compared to what it was before, but in general). Symbol servers are not geo-replicated and this wastes a lot of time if you are not deployed in West US.